src/groups/mqb/mqbs/mqbs_filebackedstorage.cpp

```diff
@@ -968,6 +1005,45 @@ void FileBackedStorage::purge(const mqbu::StorageKey& appKey)
     }
 }
 
+void FileBackedStorage::startAutoConfirming(const bmqt::MessageGUID& msgGUID)
+{
+    clearSelection();
+    d_currentlyAutoConfirming = msgGUID;
```

Comment:
It's possible to use RAII here, set `d_currentlyAutoConfirming` when a context created and call `clearSelection()` in destructor.

Reply:
Not with a stack variable.  A selection can span multiple calls after `startAutoConfirming`